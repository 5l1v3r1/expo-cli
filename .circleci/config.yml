version: 2.1

executors:
  node-10:
    docker:
      - image: circleci/node:10
    environment:
      EXECUTOR_NAME: node-10
  node-12:
    docker:
      - image: circleci/node:12
    environment:
      EXECUTOR_NAME: node-12
  puppeteer:
    docker:
      - image: circleci/node:latest-browsers
    environment:
      EXECUTOR_NAME: puppeteer
workflows:
  version: 2
  build-and-test:
    jobs:
      - build:
          name: puppeteerEnv
          e: puppeteer
      - test-webpack-config-e2e:
          e: puppeteer
          requires:
            - puppeteerEnv
jobs:
  build:
    parameters:
      e:
        type: executor
    executor: << parameters.e >>
    working_directory: ~/expo-cli
    steps:
      - checkout
      - run:
          # https://discuss.circleci.com/t/cannot-use-circle-yml-environment-variables-in-cache-keys/10994/20
          name: Configure env
          command: |
            echo "$EXECUTOR_NAME" > /tmp/.executor_name
      - restore_cache:
          name: Restore Yarn Package Cache
          keys:
            - yarn-{{ checksum "/tmp/.executor_name" }}-{{ checksum "yarn.lock" }}
      - run:
          name: Install Dependencies
          command: yarn install --frozen-lockfile
      - run:
          name: Bootstrap (Link) Internal Dependencies
          command: yarn bootstrap
      - persist_to_workspace:
          root: '~'
          paths:
            - expo-cli
      - save_cache:
          name: Save Yarn Package Cache
          key: yarn-{{ checksum "/tmp/.executor_name" }}-{{ checksum "yarn.lock" }}
          paths:
            - ~/.cache/yarn
  test-webpack-config-e2e:
    parameters:
      e:
        type: executor
    executor: << parameters.e >>
    working_directory: ~/expo-cli/packages/webpack-config
    steps:
      - attach_workspace:
          at: '~'
      - run: yarn test:e2e
