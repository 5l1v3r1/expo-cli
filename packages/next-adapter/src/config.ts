import {
  withUnimodules,
  getModuleFileExtensionsWithoutDotPrefix,
} from '@expo/webpack-config/utils';

export const pageExtensions = getModuleFileExtensionsWithoutDotPrefix('web');

export const devSwSrc = require.resolve(`@expo/webpack-config/web-default/expo-service-worker.js`);

export { withUnimodules };
