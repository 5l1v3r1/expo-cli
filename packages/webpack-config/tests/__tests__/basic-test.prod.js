/* global browser */
import lighthouse from 'lighthouse';
import url from 'url';
import config from '../../jest-puppeteer.config';

describe(`Lighthouse`, () => {
  async function getLighthouseResultsAsync(lhr, property) {
    const result = lhr.categories[property] || lhr.audits[property];

    const score = await result.score;

    // A null value is returned when the test wasn't run
    return score === null ? 1.0 : score;
  }

  const tests = [
    { field: 'performance', name: 'a performance', minValue: 0.97 },
    { field: 'accessibility', name: 'an accessibility', minValue: 0.9 },
    { field: 'best-practices', name: 'best practices', minValue: 0.93 },
    { field: 'speed-index', name: 'a page loading', minValue: 0.99 },
    { field: 'pwa', name: 'a Progressive Web App', minValue: 0.93 },
    { field: 'seo', name: 'SEO' },
    { field: 'color-contrast', name: 'a11y Contrast' },
    { field: 'image-alt', name: 'a11y Alt Text' },
    { field: 'aria-valid-attr', name: 'aria attributes' },
    { field: 'aria-valid-attr-value', name: 'aria values' },
    { field: 'tabindex', name: 'no tabIndex values above 0' },
    { field: 'logical-tab-order', name: 'logical tab order for assitive technology use' },
    { field: 'no-vulnerable-libraries', name: 'library vulnerability' },
  ];

  let lhr;
  beforeAll(async () => {
    // https://github.com/GoogleChrome/lighthouse/issues/8909#issuecomment-501992746
    global.URL = global.URL || require('url').URL;

    jest.setTimeout(60000);

    lhr = (await lighthouse(config.url, {
      port: url.parse(browser.wsEndpoint()).port,
      output: 'json',
      logLevel: 'error',
    })).lhr;
  });

  for (const lighthouseTest of tests) {
    const { minValue = 1.0 } = lighthouseTest;
    it(
      minValue >= 1.0
        ? `should have a perfect ${lighthouseTest.name} score`
        : `should have ${lighthouseTest.name} score of ${minValue * 100} or greater`,
      async () => {
        const score = await getLighthouseResultsAsync(lhr, lighthouseTest.field);
        expect(score).toBeGreaterThanOrEqual(minValue);
        expect(score).toMatchSnapshot();
      }
    );
  }
});
