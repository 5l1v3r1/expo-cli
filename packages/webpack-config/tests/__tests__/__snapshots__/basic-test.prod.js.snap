// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Lighthouse should have a Progressive Web App score of 93 or greater 1`] = `0.93`;

exports[`Lighthouse should have a page loading score of 99 or greater 1`] = `1`;

exports[`Lighthouse should have a perfect SEO score 1`] = `1`;

exports[`Lighthouse should have a perfect a11y Alt Text score 1`] = `1`;

exports[`Lighthouse should have a perfect a11y Contrast score 1`] = `1`;

exports[`Lighthouse should have a perfect aria attributes score 1`] = `1`;

exports[`Lighthouse should have a perfect aria values score 1`] = `1`;

exports[`Lighthouse should have a perfect library vulnerability score 1`] = `1`;

exports[`Lighthouse should have a perfect logical tab order for assitive technology use score 1`] = `1`;

exports[`Lighthouse should have a perfect no tabIndex values above 0 score 1`] = `1`;

exports[`Lighthouse should have a performance score of 97 or greater 1`] = `1`;

exports[`Lighthouse should have an accessibility score of 90 or greater 1`] = `1`;

exports[`Lighthouse should have best practices score of 93 or greater 1`] = `0.93`;
